opendcs-rest-api/src/main/java/org/opendcs/odcsapi/res/AppResources.java

```diff
@@ -105,112 +137,202 @@ public Response getApp(@QueryParam("appid") Long appId)
 		Optional<ApiEventClient> cli = clientConnectionCache.getApiEventClient(appId, session.getId());
 		ApiAppStatus appStat = null;
-		try (DbInterface dbi = new DbInterface();
-			ApiAppDAO dao = new ApiAppDAO(dbi))
+		try(LoadingAppDAI dai = getLegacyDatabase().makeLoadingAppDAO())
 		{
 			ApiEventClient apiEventClient = null;
-			appStat = dao.getAppStatus(appId);
-			if (appStat.getPid() == null)
+			appStat = getAppStatus(dai, appId);
+			if (appStat == null)
 			{
 				cli.ifPresent(c -> clientConnectionCache.removeApiEventClient(c, session.getId()));
-				throw new WebAppException(ErrorCodes.NO_SUCH_OBJECT, "appid " + appId 
-					+ " (" + appStat.getAppName() + ") is not running.");
+				throw new WebAppException(HttpServletResponse.SC_NOT_FOUND, "appid " + appId
+						+ " is not running (no lock found).");
+			}
+			else if (appStat.getPid() == null)
+			{
+				cli.ifPresent(c -> clientConnectionCache.removeApiEventClient(c, session.getId()));
+				throw new WebAppException(HttpServletResponse.SC_NOT_FOUND, "appid " + appId
+						+ " (" + appStat.getAppName() + ") is not running.");
 			}
 			else if (System.currentTimeMillis() - appStat.getHeartbeat().getTime() > 20000L)
 			{
 				cli.ifPresent(c -> clientConnectionCache.removeApiEventClient(c, session.getId()));
-				throw new WebAppException(ErrorCodes.NO_SUCH_OBJECT, "appid " + appId 
-					+ " (" + appStat.getAppName() + ") is not running (stale heartbeat).");
+				throw new WebAppException(HttpServletResponse.SC_NOT_FOUND, "appid " + appId
+						+ " (" + appStat.getAppName() + ") is not running (stale heartbeat).");
 			}
 			else if (!cli.isPresent())
 			{
 				Integer port = appStat.getEventPort();
 				if (port == null)
-					return ApiHttpUtil.createResponse(new ArrayList<ApiAppEvent>());
+				{
+					return Response.status(HttpServletResponse.SC_OK)
+							.entity(new ArrayList<ApiAppEvent>()).build();
+				}
 				apiEventClient = new ApiEventClient(appId, appStat.getHostname(), port, appStat.getAppName(), appStat.getPid());
 				apiEventClient.connect();
-				LOGGER.debug("Connected to {}:{}", appStat.getHostname(), port);
 				clientConnectionCache.addApiEventClient(apiEventClient, session.getId());
 			}
 			else if (appStat.getPid() != null && appStat.getPid() != cli.get().getPid())
 			{
 				// This means that the app was stopped and restarted since we last checked for events.
 				// Close the old client and open a new one with the correct PID.
 				cli.ifPresent(c -> clientConnectionCache.removeApiEventClient(c, session.getId()));
-				
+
 				Integer port = appStat.getEventPort();
 				if (port == null)
-					return ApiHttpUtil.createResponse(new ArrayList<ApiAppEvent>()); // app not running
+				{
+					return Response.status(HttpServletResponse.SC_OK)
+							.entity(new ArrayList<ApiAppEvent>()).build();
+				}
 				apiEventClient = new ApiEventClient(appId, appStat.getHostname(), port, appStat.getAppName(), appStat.getPid());
 				apiEventClient.connect();
```

Comment:
This logic is being duplicated.  Its unclear to me why the first branch doesn't need to check appStat.getPid() != null but the second branch does.  In the second branch you don't need to check cli.ifPresent it has to be present or you would be in the first branch.  I think these two could be combined and the code would be clearer.

Reply:
Cleaned up duplicated logic